### PR TITLE
test: add failing popup tests

### PR DESCRIPTION
**Pull request by dgozman**

### CODE REVIEW

File: test/browsercontext.spec.js [hunk 1]
```diff
@@ -651,4 +651,36 @@ describe('Events.BrowserContext.Page', function() {
     ]);
     await context.close();
   });
+  it.fail(CHROMIUM || WEBKIT)('should work with Shift-clicking', async({browser, server}) => {
```
Comment (yury-s) [inline test/browsercontext.spec.js:654]: Do these two tests somehow differ from those in popup.spec.js below?

File: test/popup.spec.js [hunk 2]
```diff
@@ -304,6 +304,43 @@ describe('Page.Events.Popup', function() {
     expect(await popup.evaluate(() => !!window.opener)).toBe(true);
     await context.close();
   });
+  it.fail(true)('should work with Shift-clicking', async({browser, server}) => {
+    // Chromium:
+    // - Shift+Click fires frameRequestedNavigation that never materializes
+    //   because it actually opens a new window.
+    // - New window does not report an opener.
+    // WebKit: Shift+Click does not open a new window.
```
Comment (yury-s) [inline test/popup.spec.js:312]: I feel like it may be platform specific, let me check on linux.

**Review by yury-s — APPROVED** [review]

File: test/popup.spec.js (hunk 2 above)
Comment (yury-s) [reply test/popup.spec.js:312]: It's loading new page in the same window on GTK :/

**Review by yury-s — APPROVED** [review]

File: test/browsercontext.spec.js (hunk 1 above)
Comment (dgozman) [reply test/browsercontext.spec.js:654]: This one waits for the 'page' event, the other waits for 'popup' event.

File: test/popup.spec.js (hunk 2 above)
Comment (dgozman) [reply test/popup.spec.js:312]: Same for me on mac.

**Pull request merged**